components/sidebar/sidebar_channel/channel_pencil_icon/index.ts

```diff
@@ -0,0 +1,32 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {connect} from 'react-redux';
+
+import {getCurrentChannelId} from 'mattermost-redux/selectors/entities/channels';
+import {Channel} from 'mattermost-redux/types/channels';
+
+import {getPostDraft} from 'selectors/rhs';
+import {StoragePrefixes} from 'utils/constants';
+import {GlobalState} from 'types/store';
+
+import ChannelPencilIcon from './channel_pencil_icon';
+
+type OwnProps = {
+    channel?: Channel;
+}
+
+function hasDraft(draft: any, currentChannelId?: string, channel?: Channel) {
+    return draft && Boolean(draft.message.trim() || draft.fileInfos.length || draft.uploadsInProgress.length) && currentChannelId !== channel?.id;
+}
+
+function mapStateToProps(state: GlobalState, ownProps: OwnProps) {
+    const currentChannelId = getCurrentChannelId(state);
+    const draft = ownProps.channel?.id ? getPostDraft(state, StoragePrefixes.DRAFT, ownProps.channel.id) : null;
+
+    return {
+        draft: hasDraft(draft, currentChannelId, ownProps.channel) ? draft : null,
```

Comment:
Could we switch this back to just passing a boolean? That'll save the `ChannelPencilIcon` from re-rendering whenever the draft changes since it doesn't care about what's actually in the draft.

Also, I really like the change of where we're putting the draft icon. This seems a lot nicer than replacing the channel icon.

Reply:
We were going to add a tooltip upon hovering the icon to show a preview of the draft message, 
but we can do it later when we implement that change